src/main/scala/com/redislabs/provider/redis/rdd/RedisRDD.scala

```diff
@@ -50,8 +50,8 @@ class RedisKVRDD(prev: RDD[String],
         .flatMap{
           // Silently filter out this exception, when the value doesn't match the expected type "String"
           case (_, e: JedisDataException) if Option(e.getMessage).getOrElse("").contains("WRONGTYPE") => None
-            // Throw any other JedisDataException we encounter
-          case (_, e: JedisDataException) => throw e
+            // Throw any other Exception we encounter
+          case (_, e: Throwable) => throw e
```

Comment:
@fe2s changed this. Please review this change again.

Reply:
Thanks @mayankasthana . Can we please also extract this parsing into the `ParseUtils`? I would keep this logic together in a single class, so it will easier to make any changes in future. Other than that it looks good.

Reply:
Done